nixos/modules/system/boot/timesyncd.nix

```diff
@@ -82,8 +93,9 @@ with lib;
 
     environment.etc."systemd/timesyncd.conf".text = ''
       [Time]
-      NTP=${concatStringsSep " " config.services.timesyncd.servers}
-      ${config.services.timesyncd.extraConfig}
+      NTP=${concatStringsSep " " cfg.servers}
```

Comment:
Wouldn't setting this to an empty string imply the defaults are always used, and it's not possible to receive config over DHCP?

I feel like we need a VM test that sends config over DHCP and peeks at what's getting set

Reply:
> > When the empty string is assigned, the list of NTP servers is reset, and all prior assignments will have no effect.

This probably refers to dropins like for service files. eg. if you want to overwrite ExecStart= you first need to set it to empty otherwise systemd thinks it is a list which doesn't work.

Reply:
> ```shell
> FallbackNTP=0.nixos.pool.ntp.org 1.nixos.pool.ntp.org 2.nixos.pool.ntp.org 3.nixos.pool.ntp.org
> ```

This is already an implied default because of https://github.com/NixOS/nixpkgs/blob/3526bfa2358f19720437a9e0bbf4a302a2f943cb/pkgs/os-specific/linux/systemd/default.nix#L664 https://github.com/systemd/systemd/blob/0197fb599ac4f29871e8ea1923be7b14bbd7bbf0/src/timesync/timesyncd.conf.in#L21



> or remove the empty `NTP=` altogether when `servers` is the empty list:

I think we should remove it because we do not intent do explicitly remove any previously set value.